### PR TITLE
Add LTI 1.1 config handler for the /lti11/config route

### LINKED ISSUE #49: Add an LTI 1.1 Tool Provider XML endpoint
### Proposed change

Add a new LTI 1.1 handler to dynamically display the LTI 1.1 configuration XML.

### Alternative options

To add an LTI Link for a [Tool Consumer](https://www.imsglobal.org/specs/lti/xml) users have to manually include the LTI 1.1 settings values to the LMS' LTI 1.1 configuration page, manually create an XML based on the standard LTI 1.1 specification to paste it into the LMS configuration page, among others.

### Who would use this feature?

Any user that needs to quickly add the JupyterHub as a Tool Provider to a Tool Consumer (LMS).

### Suggest a solution

Create a handler and publish the correct XML with a web-accessible endpoint based on JupyterHub's URL, placements settings, etc.

### PR DESCRIPTION
**Pull request by jgwerner**
- Adds the LTI 1.1 configuration handler to render the authenticator's configuration in XML format
- Updates README.md to included common settings section and adds LTI 1.1 configuration endpoint settings

Closes #49

### CODE REVIEW

File: ltiauthenticator/lti11/auth.py
Comment (consideRatio) [inline ltiauthenticator/lti11/auth.py:48]: I assume this is the unicode string to a public URL? Is that correct? Perhaps reasonable to be more explicit about that. I don't know if this is obvious or not given the context though, but I guess it could be useful to be explicit without knowing.

**Comment by consideRatio** [comment]
@jagwar from what I can tell given no actual experience with LTI, this LGTM.

File: ltiauthenticator/lti11/auth.py
Comment (jgwerner) [reply ltiauthenticator/lti11/auth.py:48]: Good point, I'll update the help content with a clarification on how to reference the icon image.

File: README.md
Comment (consideRatio) [inline README.md:37]: This mentions edX while under a common system section.

File: README.md
Comment (consideRatio) [inline README.md:46]: These relate to the section and should be indented.

File: README.md
Comment (consideRatio) [inline README.md:48]: Numbering is now weird, you can use 1. at all time to let the rendered markdown figure out the numbers to avoid these kinds of troubles.

File: README.md
Comment (consideRatio) [inline README.md:62]: Comments that seem to relate to the above configuration step in the main bullet point list, so should perhaps be indented also?

File: README.md
Comment (consideRatio) [inline README.md:86]: Indentation inconsistencies and double an uneven amount of `"` used making it invalid yaml.

File: ltiauthenticator/lti11/handlers.py
Comment (consideRatio) [inline ltiauthenticator/lti11/handlers.py:60]: Something like `Renders` instead of `Gets` perhaps? It writes back a response to the HTTP request of GET type.

```suggestion
        Renders the XML config which is used by LTI consumers to install the external tool.
```

**Review by consideRatio — COMMENTED** [review]
This looks good to me but there were some issues with the markdown in the README.md after merging some formatting PR I think =/

Let's stop using explicit numbers and use `1.` at all time to avoid numbering issues and make sure all numbered bullet points have their content indented under it.

**Comment by jgwerner** [comment]
Hi, @consideRatio thanks for the detailed review comments regarding the docs. I found some issues when testing with our Canvas sandbox and am updating the handlers/tests so will make sure to update the docs as well with the next push.

**Comment by jgwerner** [comment]
@consideRatio I'm not sure if you follow the Semantic Release convention or not (I think @yuvipanda once mentioned that you do not). If you do, then this PR adds a new feature so it should require a new minor release version. If not, your call!

File: README.md
Comment (consideRatio) [inline README.md:49]: This section is confusing to me.

1. "There are two configuration values to define with LTI 1.1:"
   I wonder both about "two" and "to define". Is the user required to set two values? Then why do I only see one listed as required in the table below?
2. There is a bullet list table, followed by a markdown formatted table and both include the same things. I assume the bullet list table should be removed as an outdated duplicate.

File: README.md
Comment (consideRatio) [inline README.md:110]: The documentation above is outdated, it should be like this since ~0.10.X or 0.11.X i think, and the chart is now at 1.1.3.

```suggestion
# Custom config for JupyterHub's helm chart
hub:
  config:
    # Additional documentation related to authentication and authorization available at
    # https://zero-to-jupyterhub.readthedocs.io/en/latest/administrator/authentication.html
    JupyterHub:
      authenticator_class: lti
    LTI11Authenticator:
      consumers: { "client-key": "client-secret" }
      username_key: "lis_person_contact_email_primary"
      config_icon: "https://my.static.assets/img/icon.jpg"
```

File: README.md
Comment (consideRatio) [inline README.md:113]:
```suggestion
> _Note_: Configuring for example `hub.config.LTI11Authenticator.username_key=lis_person_contact_email_primary` above is the same as in a `jupyterhub_config.py writing `c.LTI11Authenticator.username_key = lis_person_contact_email_primary`.
```

**Review by consideRatio — COMMENTED** [review]
I made some documentation suggestions, but otherwise this LGTM from what I can tell.

**Comment by consideRatio** [comment]
How is the endpoint `/lti11/config` that is returning the rendered XML template used in practice? Is it required to be around by LTI v1.1 or similar for example? Perhaps you can answer by updating the PR description for anyone reading info about this PR later as well.

File: ltiauthenticator/lti11/handlers.py
Comment (consideRatio) [inline ltiauthenticator/lti11/handlers.py:56]: Could you elaborate a bit in this docstring, perhaps link out to relevant schemas as described in #49?

File: README.md
Comment (jgwerner) [reply README.md:49]: You're right, I goofed and had left the bullets in there by mistake. Only the table is documented now.

File: README.md
Comment (jgwerner) [reply README.md:110]: Corrected.

**Comment by jgwerner** [comment]
@consideRatio updates are pushed based on your suggestions

**Review by consideRatio — APPROVED** [review]
Wieee nice! Ready for merge I'd say, do you want to review this also @yuvipanda?

**Comment by yuvipanda** [comment]
Go for it, @consideRatio. Let's be generous with our minor versions :)

**Comment by consideRatio** [comment]
Wieeee great work as always @jgwerner, I appreciate how caring you are for this project!!

**Pull request merged**
